base/compiler/abstractinterpretation.jl

```diff
@@ -1129,11 +1159,11 @@ function abstract_call_known(interp::AbstractInterpreter, @nospecialize(f),
         # mark !== as exactly a negated call to ===
         rty = abstract_call_known(interp, (===), fargs, argtypes, sv).rt
         if isa(rty, Conditional)
-            return CallMeta(Conditional(rty.var, rty.elsetype, rty.vtype), nothing) # swap if-else
+            return CallMeta(Conditional(rty.var, rty.elsetype, rty.vtype), false) # swap if-else
         elseif isa(rty, Const)
-            return CallMeta(Const(rty.val === false), nothing)
+            return CallMeta(Const(rty.val === false), MethodResultPure())
```

Comment:
definitely not pure

Reply:
because `rty` used to have actual provenance, but just threw it away?

Reply:
ah, unfolding context in github further, I see contextually we're guessing that this is the only legal answer from `!==`?

Reply:
Nope, I'm still confused. For builtins that are in `_PURE_BUILTINS` and return `Const` we just implicitly treat the result as pure. That doesn't apply here, because `!==` is not a builtin, but we are treating it as if it were one that simply had its result inverted. Why does that invalidate the purity of the call?

Reply:
jinx with your previous comment, but yes, `!==` is special.